services/UserActivityService.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const RestApiModel = require('../models/RestApiModel');
+const { DateTime } = require('luxon');
+class UserActivityService {
+
+  /**
+   * Calls the REST API with the given options and returns an array of search results as a Promise.
+   *
+   * @param options
+   * @returns {Promise<[record]>}
+   */
+  static async searchByParams(options) {
+    return await new Promise((resolve, reject) => {
+      const model = new RestApiModel({}, options);
+      model.fetch((err, data, _responseTime) => {
+        if (err) {
+          reject(err);
+        } else {
+          Object.keys(data).length !== 0 ? resolve(this.processRecord(data, options.searchParams, options.includeWeekends)) : resolve(data);
+        }
+      });
+    });
+  }
+
+  /**
+   * Transform/decorate the record with additional properties
+   *
+   * @param record
+   * @param params
+   * @param includeWeekends
+   * @returns {{header: {data: string, isWeekend}[], rows: *[]}}
+   */
+  static processRecord(record, params, includeWeekends) {
+    const from = DateTime.fromISO(params.from);
+    const to = DateTime.fromISO(params.to);
+    const diff = to.diff(from, 'days').days;
+
+    // create array of dates between to and from user input and whether the date is a weekend
+    let dateArr = Array(diff).fill(undefined).map((i, index) => {
+      const day = from.plus({ days: index });
+      return { data: day.toFormat('dd-MM-yyyy'), isWeekend: day.toFormat('cccc') === 'Saturday' || day.toFormat('cccc') === 'Sunday' };
```

Comment:
Yeah, not sure that's an improvement.

I was thinking more like
const SAT = 6;
const SUN = 7;
day.weekday === SAT || day.weekday === SUN

Otherwise it appears there's a weekdayShort()

day.weekdayShort === 'Sat' || day.weekdayShort === 'Sun'

Reply:
Setting constants to use the weekday index in Luxon doesn't really add anything IMO. 

I've amended to the latter suggestion but seems much of a muchness to me.